File: ShoppingList/src/main/java/org/openintents/shopping/BaseOptionalDependencies.java
package org.openintents.shopping;

import android.app.Activity;
import android.content.Context;

import org.openintents.shopping.sync.NoSyncSupport;
import org.openintents.shopping.ui.ToggleBoughtInputMethod;
import org.openintents.shopping.ui.widget.ShoppingItemsView;

/**
 * This is the default implementation for all product flavors for any implementation
 * <p>
 * If the signature is changed make sure that the corresponding implementations are changed
 * because Android Studio does only show usage for the current build flavor.
 */
public class BaseOptionalDependencies {

    public void onResumeShoppingActivity(Activity context) {
        // do nothing;
    }

    public ToggleBoughtInputMethod getToggleBoughtInputMethod(Context context, ShoppingItemsView itemsView) {
        return null;
    }

    public SyncSupport getSyncSupport(final Context context) {
        return new NoSyncSupport();
    }
}

File: ShoppingList/src/main/java/org/openintents/shopping/BaseOptionalDependencies.kt
package org.openintents.shopping

import android.app.Activity
import android.content.Context
import org.openintents.shopping.sync.NoSyncSupport
import org.openintents.shopping.ui.ToggleBoughtInputMethod
import org.openintents.shopping.ui.widget.ShoppingItemsView

/**
 * This is the default implementation for all product flavors for any implementation
 *
 * If the signature is changed make sure that the corresponding implementations are changed
 * because Android Studio does only show usage for the current build flavor.
 */
open class BaseOptionalDependencies {

    open fun onResumeShoppingActivity(context: Activity) {
        // do nothing
    }

    open fun getToggleBoughtInputMethod(context: Context, itemsView: ShoppingItemsView): ToggleBoughtInputMethod? {
        return null
    }

    open fun getSyncSupport(context: Context): SyncSupport {
        return NoSyncSupport()
    }
}
